client/src/js/services/journalGridService.js

```diff
@@ -0,0 +1,338 @@
+    gridService.columnsService = columnsService;
+    gridService.managerService = managerService;
+    gridService.dataLoaderService = dataLoaderService;
+    gridService.idDom = '#journal_grid';
+    gridService.sortColumn = null;
+    gridService.options = {
+        enableCellNavigation: true,
+        enableColumnReorder: true,
+        forceFitColumns: true,
+        editable: true,
+        rowHeight: 30,
+        autoEdit: false
+    };
+
+    gridService.buildGrid = function buildGrid (){
+      gridService.grid = new Slick.Grid(gridService.idDom, gridService.dataviewService.dataview, gridService.columnsService.columns, gridService.options);
+      gridService.grid.setSelectionModel(new Slick.RowSelectionModel({selectActiveRow: false}));
+      gridService.grid.registerPlugin(gridService.dataviewService.gimp);
+      return gridService.grid;
+    };
+
+    gridService.subscribeToOnCellChange = function subscribeToOnCellChange (){
+      if(!gridService.dataviewService.dataview) {throw 'undefined dataview';}
+      if(!gridService.grid) {throw 'undefined grid, call buildGrid method first!';}
+
+      gridService.grid.onCellChange.subscribe(function(e, args) {
+        var id = args.item.id || args.item.uuid;
+        gridService.dataviewService.dataview.updateItem(id, args.item);
+      });
+    }; 
+
+    gridService.subscribeToOnSort = function subscribeToOnSort (){
+      if(!gridService.dataviewService.dataview) {throw 'undefined dataview';}
+      if(!gridService.grid) {throw 'undefined grid, call buildGrid method first!';}
+
+      gridService.grid.onSort.subscribe(function (e, args) {
+        gridService.sortColumn = args.sortCol.field;
+        gridService.dataviewService.dataview.sort(sort, args.sortAsc);
+      });
+    };
+
+    gridService.subscribeToOnBeforeEditCell = function subscribeToOnBeforeEditCell (){
+      gridService.grid.onBeforeEditCell.subscribe(function (e, args) {
+        var item =  gridService.dataviewService.dataview.getItem(args.row);
+        if (gridService.managerService.getMode() !== 'edit' || gridService.managerService.getSessionTransactionId() !== item.trans_id ) { return false; }
+      });
+    };     
+
+    gridService.subscribeToOnclick = function subscribeToOnclick (){
+      if(!gridService.grid) {throw 'undefined grid, call buildGrid method first!';}
+      gridService.grid.onClick.subscribe(function (e, args) {
+        handleClick(e.target.className, args);
+      });
+    };
+
+
+    function handleClick(className, args) {
+      var classes = className.split(' ');
+      var buttonMap = {
+        'editTransaction'   : editTransaction,
+        'deleteRow'         : deleteRow,
+        'addRow'            : addRow,
+        'saveTransaction'   : saveTransaction
+        // 'deleteTransaction' : deleteTransaction
+      };
+      classes.forEach(function (cls) {
+        if (buttonMap[cls]) { buttonMap[cls](args); }
+      });
+    }
+
+    function editTransaction(args) {
```

Comment:
Would it be possible to have a `GridTransactionService` to handle the creation of transactions?  It would take care of things like:
 1. Create the initial template
 2. TransactionService.addRow(), TransactionService.removeRow(idx)
 3. The validation logic could be contained either in the TransactionService or in a separate TransactionValidationService.

I think that would help reduce the complexity in both this controller and managerService.